vfs_test.go

```diff
@@ -608,3 +609,34 @@ func TestOS_ReadLink_ErrOutsideRoot(t *testing.T) {
 		t.Errorf("%s: ReadLink: got %q, want %q", osfs, dst, want)
 	}
 }
+
+func TestUnion(t *testing.T) {
+	m1 := map[string]string{"foo/file": ""}
+	m2 := map[string]string{"bar/file": ""}
+	u := rwvfs.Union(rwvfs.Map(m1), rwvfs.Map(m2))
+
+	infos, err := u.ReadDir("/")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if got, want := len(infos), 2; got != want {
+		t.Errorf(`ReadDir: got %d, want %d`, got, want)
+	}
+
+	testCreate(t, u, "test", m1)
+	testCreate(t, u, "foo/test", m1)
+	testCreate(t, u, "bar/test", m2)
+}
+
+func testCreate(t *testing.T, u rwvfs.FileSystem, path string, m map[string]string) {
+	w, err := u.Create(path)
+	if err != nil {
+		panic(err)
+	}
+	w.Write([]byte("test"))
+	w.Close()
+
+	if got, want := m[path], "test"; got != want {
+		t.Errorf(`Create: got %d, want %d`, got, want)
```

Comment:
An incorrect verb `%d` is being used here for type `string`. Either `%s` or the more general `%v` should be used instead.